src/backend/distributed/metadata/distobject.c

```diff
@@ -160,12 +184,28 @@ ObjectExists(const ObjectAddress *address)
 void
 MarkObjectDistributed(const ObjectAddress *distAddress)
 {
+	MarkObjectDistributedWithName(distAddress, "");
+}
+
+
+/*
+ * MarkObjectDistributedWithName marks an object as a distributed object.
+ * Same as MarkObjectDistributed but this function also allows passing an objectName
+ * that is used in case the object does not exists for the current transaction.
+ */
+void
+MarkObjectDistributedWithName(const ObjectAddress *distAddress, char *objectName)
+{
+	if (!CitusHasBeenLoaded())
```

Comment:
Sorry for the ping-pong response. But I realized my previous comment on this did not make sense.

> I feel like if CitusHasBeenLoaded() returns true we should still mark this object as distributed, even if the current database is not marked as the main database.

We cannot mark something as distributed in the non-main database. I do think we should do both though. We should check if the current database is the main database AND if CitusHasBeenLoaded.

Reply:
never mind this, the current check is correct. Monday morning brain fart.